### PR TITLE
Update Netwonsoft.Json dependency

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [ ] ~Are tests passing locally?~
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests if this modifies the Windows code?~
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Related PRs or issues

[AB#77598](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/77598)

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -741,8 +740,7 @@ void BuildXcodeProject(string projectPath)
 }
 
 // Download json package.
-Task("Externals-Uwp-IL2CPP-Dependencies-BuildNewtonsoftJson")
-    .Does(() =>
+Task("BuildNewtonsoftJson").IsDependentOn("Externals-Uwp-IL2CPP-Dependencies").Does(() =>
```
Comment (MatkovIvan) [inline build.cake]: vice versa. `Externals-Uwp-IL2CPP-Dependencies` depends on `BuildNewtonsoftJson`. Currently it doesn't run

File: build.cake [hunk 2]
```diff
@@ -735,4 +736,37 @@ void BuildXcodeProject(string projectPath)
...
+    var packageSource = $"{progectPath}/bin/Release/unity-aot/{packageName}";
+
+    Information("Delete the assemble in case it already exists in the Assets folder.");
+    var existingPackage = $"{packageDestination}/{packageName}";
+    if (FileExists(existingPackage))
+    {
+        DeleteFile(existingPackage);
+    }
+
+    Information($"Move assemble to {packageDestination} path.");
```
Comment (annakocheshkova) [inline build.cake]:
```suggestion
    Information($"Moving assembly to {packageDestination} path.");
```

**Review by annakocheshkova — APPROVED** [review]

File: build.cake [hunk 3]
```diff
@@ -735,4 +736,37 @@ void BuildXcodeProject(string projectPath)
     });
 }
 
+// Download json package.
+Task("BuildNewtonsoftJson").Does(() =>
+{
+    if (IsRunningOnUnix()) {
+        Information("Skip building newtonsoft task on unix.");
```
Comment (MatkovIvan) [inline build.cake]: All UWP actions should be ignored

File: build.cake [hunk 4]
```diff
@@ -735,4 +736,37 @@ void BuildXcodeProject(string projectPath)
     });
 }
 
+// Download json package.
```
Comment (MatkovIvan) [inline build.cake]: Build

File: build.cake [hunk 5]
```diff
@@ -735,4 +736,37 @@ void BuildXcodeProject(string projectPath)
...
+        return;
+    }
+
+    var packageDestination = "Assets/AppCenter/Plugins/WSA/IL2CPP";
+    var progectPath = "Modules/Newtonsoft.Json-for-Unity/Src/Newtonsoft.Json";
+    var pathSolution = $"{progectPath}/Newtonsoft.Json.csproj";
+
+    Information("Build project.");
+    MSBuild(pathSolution, c => c
+    .WithRestore()
```
Comment (MatkovIvan) [inline build.cake]: Format

File: cgmanifest.json [hunk 6]
```diff
@@ -37,15 +37,6 @@
                 }
             }
         },
-        {
```
Comment (MatkovIvan) [inline cgmanifest.json:40]: Add as git reference instead

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: .gitmodules [hunk 7]
```diff
@@ -1,3 +1,6 @@
 [submodule "BreakpadSupport/breakpad-support/src/main/cpp/google-breakpad"]
 	path = BreakpadSupport/breakpad-support/src/main/cpp/google-breakpad
 	url = https://github.com/microsoft/appcenter-sdk-android-breakpad.git
+[submodule "Modules/Newtonsoft.Json-for-Unity"]
+	path = Modules/Newtonsoft.Json-for-Unity
+	url = https://github.com/jilleJr/Newtonsoft.Json-for-Unity.git
```
Comment (iamclement) [inline .gitmodules:6]: We'll need to add an entry in the ThirdPartyNotices.txt for this one as it's distributed as part of the SDK.

**Review by iamclement — CHANGES_REQUESTED** [review]

**Review by annakocheshkova — APPROVED** [review]

**Review by MatkovIvan — APPROVED** [review]

**Review by guperrot — APPROVED** [review]

**Review by iamclement — APPROVED** [review]

**Pull request merged**